src/winforms/toga_winforms/command.py

```diff
@@ -13,3 +13,12 @@ def __init__(self, interface):
                 self.interface.icon = CoreIcon(self.interface.icon_id)
         else:
             self.interface.icon = None
+
+    @property
+    def enabled(self):
+        return self.interface.enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        for widget in self.interface._widgets:
+            widget.Enabled = self.interface.enabled
```

Comment:
Are you certain this part is needed? The interface level enabled() should already be iterating over the widgets... won't this be invoking (widget interface).enabled twice?

Reply:
The interface level enabled() is iterating over the widgets' native objects, so it sends `enabled = True` to the `Winforms.MenuStripItem`, which doesn't have `enabled` property, it only has `Enabled` one. So, the interface level call doesn't do anything. And because the toolbar/menubar items aren't separate widgets, I don't know how to make the interface call `enabled = True` turn into `Enabled = True`.

Reply:
Ah - I hadn't caught that at the interface level, `_widgets` is a list of native objects. That also explains why the GTK implementation is broken (see #722); it's an accident that Cocoa uses the `.enabled` property, so it works there.

That's the root cause of the problem - we've got native objects being exposed in the interface layer. We should be wrapping those native objects in an impl so that we can use a common API call (at the impl level) to trigger the native behavior. I'd imagine each platform having a wrapper class something like:
```
class CommandWidget:
    def __init__(self, native):
        self.native = native
    def set_enabled(self, value):
        self.native.enabled = value
```

The Windows version would have `self.native.Enabled`; GTK would have a call to `self.native.set_sensitive()` (or whatever is appropriate). 

That removes the need for impl-level handling on the Command itself; and if we ever end up with a platform where the mechanism to enable/disable a menu item is different to a toolbar button, we just need a different impl wrapper object.

Does that make sense?

Reply:
That makes sense.
So, what we need to do is to create a new module in `widgets` folder for `CommandWidget`. I'm a little confused as to why `__init__` takes `native` as an argument. Usually, the argument passed to widgets is the `interface`. Should `CommandWidget` maybe take both `native` and `interface`?

Reply:
I don't think a new module is needed; this is going to be very tightly coupled to Commands, so putting it in `command.py` for a given platform would be fine.

The reason the constructor arguments are a little different is because the `CommandWidget` widget itself is unusual. There isn't an interface layer for this widget in this case. The native widgets involved are menu items and toolbar buttons, and they aren't surfaced with an interface - they're created implicitly in response to other widgets in the user interface (apps and windows). 

Thinking about the problem a little more, a more consistent way to handle this would be to move `_widgets` from the interface Command and onto the impl for each platform (probably renaming it to `native` along the way, since that's what they really are - the native manifestation of the Command). That would be more consistent with other widgets that are composites (e.g., a Toga webview interface that is composed of a scroll view and a web canvas on some implementations).

Then, the *interface* for Command would have the enable property, invoking set_enabled() on the impl; and then the impl would iterate over the native widgets that are used to expose the Command on a given platform. The iteration pattern will be repeated on each platform impl for Command, but each platform can use the appropriate enable/disable call to affect the widget. It also means we don't need to introduce an new type of impl object for a widget that doesn't have an interface.

Reply:
I tried to implement native widget handling in the implementation layer, setting `native` to an empty list at first. This naming seems off, though, because in all other widgets, etc, native refers to one object. 
I also adjusted the tests for Command, but I couldn't find a way of testing if a native widget was added to `Command._impl.native` or not.